### PR TITLE
Fix license hashes in api settings values

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [PropertiesDao.java#L92](https://github.com/SonarSource/sonarqube/blob/cc7d54d6ed2bab44937c5ffba5e714fdcde61520/sonar-db/src/main/java/org/sonar/db/property/PropertiesDao.java#L92): Close this "PreparedStatement". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2095)
1. ![MAJOR][MAJOR] [PropertiesDao.java#L171](https://github.com/SonarSource/sonarqube/blob/cc7d54d6ed2bab44937c5ffba5e714fdcde61520/sonar-db/src/main/java/org/sonar/db/property/PropertiesDao.java#L171): Make "selectByKeys" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: server/sonar-server/src/main/java/org/sonar/server/setting/ws/ScannerSettings.java [hunk 1]
```diff
@@ -39,31 +39,25 @@
...
-  private Set<String> scannerSettingKeys;
-
-  public ScannerSettings(PropertyDefinitions propertyDefinitions, PluginRepository pluginRepository) {
+  public ScannerSettings(DbClient dbClient, PropertyDefinitions propertyDefinitions) {
+    this.dbClient = dbClient;
     this.propertyDefinitions = propertyDefinitions;
-    this.pluginRepository = pluginRepository;
   }
 
-  @Override
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/setting/ws/ScannerSettings.java]: why is it not initialized at startup? what data is not available? can be changed?

File: sonar-db/src/main/resources/org/sonar/db/property/PropertiesMapper.xml [hunk 2]
```diff
@@ -181,6 +181,17 @@
     </where>
   </select>
 
+  <select id="selectGlobalPropertiesByKeyQuery" resultType="ScrapProperty">
+    select
+    <include refid="columnsToScrapPropertyDto"/>
```
Comment (teryk) [inline sonar-db/src/main/resources/org/sonar/db/property/PropertiesMapper.xml]: what is this columnsToScrapPropertyDto?

File: sonar-db/src/main/java/org/sonar/db/property/PropertiesDao.java [hunk 3]
```diff
@@ -170,6 +172,10 @@ public PropertyDto selectProjectProperty(DbSession dbSession, long componentId,
     return executeLargeInputs(keys, partitionKeys -> getMapper(session).selectByKeys(partitionKeys, componentId));
   }
 
+  public List<PropertyDto> selectGlobalPropertiesByKeyQuery(DbSession session, String keyQuery) {
+    return getMapper(session).selectGlobalPropertiesByKeyQuery(buildLikeValue(keyQuery, WildcardPosition.BEFORE));
```
Comment (teryk) [inline sonar-db/src/main/java/org/sonar/db/property/PropertiesDao.java]: you might want to check here that keyQuery is not null

Comment (teryk) [inline sonar-db/src/main/java/org/sonar/db/property/PropertiesDao.java]: why only a WildcartPosition.BEFORE? if it's specifically for licences, then maybe you should rename to make it explicit.
my point is as it is I don't see any reason to have a key query with only a wildcard as a prefix

File: server/sonar-server/src/test/java/org/sonar/server/setting/ws/ValuesActionTest.java [hunk 4]
```diff
@@ -104,10 +98,6 @@
 
   @Before
   public void setUp() throws Exception {
-    PluginInfo pluginInfo = mock(PluginInfo.class);
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/setting/ws/ValuesActionTest.java]: you don't add a test to validate that it returns the desired settings?

File: server/sonar-server/src/main/java/org/sonar/server/setting/ws/ScannerSettings.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/setting/ws/ScannerSettings.java]: It was a big mistake to load settings at startup, as settings value can change at any time !
I'll try to add ITs BTW to cover this.

File: sonar-db/src/main/resources/org/sonar/db/property/PropertiesMapper.xml (hunk 2 above)
Comment (julienlancelot) [inline sonar-db/src/main/resources/org/sonar/db/property/PropertiesMapper.xml]: No idea

File: sonar-db/src/main/java/org/sonar/db/property/PropertiesDao.java (hunk 3 above)
Comment (julienlancelot) [inline sonar-db/src/main/java/org/sonar/db/property/PropertiesDao.java]: Indeed, it was for perf reason, but as it's not explicit I will use a BEFORE_AND_AFTER position

File: server/sonar-server/src/test/java/org/sonar/server/setting/ws/ValuesActionTest.java (hunk 4 above)
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/setting/ws/ValuesActionTest.java]: It's already there

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [PropertiesDao.java#L92](https://github.com/SonarSource/sonarqube/blob/62c010110229b99c4125a961453deadbd45d6814/sonar-db/src/main/java/org/sonar/db/property/PropertiesDao.java#L92): Close this "PreparedStatement". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2095)
1. ![MAJOR][MAJOR] [PropertiesDao.java#L171](https://github.com/SonarSource/sonarqube/blob/62c010110229b99c4125a961453deadbd45d6814/sonar-db/src/main/java/org/sonar/db/property/PropertiesDao.java#L171): Make "selectByKeys" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request merged**